app/experimenter/experiments/models/nimbus.py

```diff
@@ -301,6 +301,42 @@ def results_ready(self):
             )
             return datetime.date.today() >= resultsReadyDate
 
+    @property
+    def review_request(self):
+        changelog_event = self.changes.latest_review_request()
+        if changelog_event is None or changelog_event.new_status != self.status:
+            return None
+        return changelog_event
+
+    def _enforce_review_changelog_sequence(
+        self, previous_changelog_event, changelog_event
+    ):
+        if (
+            previous_changelog_event is None
+            or changelog_event is None
+            or previous_changelog_event.changed_on > changelog_event.changed_on
+        ):
+            return None
+        return changelog_event
```

Comment:
Yeah I see what you're getting at here, but I think there's a simpler way to do this.

Down in the manager, just

1) get only the most recent changelog
2) check if it's a review request
3) if it is, return it, otherwise return None

rathern than doing this on the instance and comparing statuses and dates, etc

That should accomplish the same thing in less work, no?